### DISCUSSION TITLE
Wrong layout when mixing equation layout and underbrace.

### DISCUSSION THREAD

**Discussion by remimimimimi** [typst/typst#4376] [General]
### Description

Same as title. Code to reproduce issue:
```typst
#set page(width: auto, height: auto, margin: 0.5cm)
let $h = underbrace(k compose p_2, "composition is a PRF of type (iv)")$ and $underbrace(k, "successor, PRF (i)") compose p_2 (m, f(m, n), n) &= k(f(m, n))\ &= f(m, n)'$
```
![test](https://github.com/typst/typst/assets/33205215/55a2f374-a983-4ba7-a693-3a0a9a493246)

### Operating system

Web app, Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
What part is wrong? This looks correct to my eyes at first glance.

**Reply by remimimimimi** [reply]
Let should be on the first line

**Reply by remimimimimi** [reply]
I actually think it shouldn't be allowed to use equation alignment in inline math mode.

**Reply by Enivex** [reply]
> Let should be on the first line

There is only one line in your picture

**Reply by Enivex** [reply]
> I actually think it shouldn't be allowed to use equation alignment in inline math mode.

I agree that you _shouldn't_ use it inline, but disallowing it may be a step too far.

**Reply by remimimimimi** [reply]
Then maybe we should throw warning at least?

**Reply by Enivex** [reply]
> Then maybe we should throw warning at least?

But it's doing what the user asked for?

**Reply by remimimimimi** [reply]
Not quite, as I would like to see `f(m, n)'` part visually on the bottom line. Is there a way to achieve that?
